include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -0,0 +1,508 @@
+    }
+
+    /*!\brief Adds a flag to the seqan3::argument_parser.
+     *
+     * \param value The variable in which to store the given command line argument.
+     * \param short_id The short identifier for the flag (e.g. `'i'`).
+     * \param long_id The long identifier for the flag (e.g. `"integer"`).
+     * \param desc The description of the flag to be shown in the help page.
+     * \param spec Advanced flag specification. see seqan3::option_spec.
+     *
+     * \throws seqan3::parser_design_error
+     */
+    void add_flag(bool & value,
+                  char const short_id,
+                  std::string const & long_id,
+                  std::string const & desc,
+                  option_spec const & spec = option_spec::DEFAULT)
+    {
+        if(id_exists(short_id))
+            throw parser_design_error("Option Identifier '" + std::string(1, short_id) + "' was already used before.");
+        if(id_exists(long_id))
+            throw parser_design_error("Option Identifier '" + long_id + "' was already used before.");
+        if (short_id == '\0' && long_id.empty())
+            throw parser_design_error("Option Identifiers cannot both be empty.");
+
+        std::visit([=, &value] (auto & f) {f.add_flag(value, short_id, long_id, desc, spec);}, format);
+    }
+
+    /*!\brief Adds a positional option to the seqan3::argument_parser.
+     *
+     * \param value The variable in which to store the given command line argument.
+     * \param desc The description of the positional option to be shown in the help page.
+     * \param validator The validator applied to the value after parsing (callable).
+     *
+     * \throws seqan3::parser_design_error
+     */
+    template <typename option_t, typename validator_type = detail::default_validator>
+    //requires is_stream_convertible_concept<option_t> || is_stream_convertible_concept<typename option_t::value_type>
+    void add_positional_option(option_t & value,
+                               std::string const & desc,
+                               validator_type && validator = validator_type())
+    {
+        std::visit([=, &value] (auto & f) {f.add_positional_option(value, desc, validator);}, format);
+    }
+    //!\}
+
+    /*!\brief Initiates the actual command line parsing.
+     *
+     * \attention The function must be called at the very end of all parser
+     * related code and should by enclosed in a try catch block.
+     *
+     * \throws seqan3::parser_interruption
+     *
+     * \details
+     *
+     * When no specific key words are supplied, the seqan3::argument_parser 
+     * starts to process the command line for specified options, flags and
+     * positional options.
+     *
+     * The parser behaves differently when the the given command line (argv) 
+     * contains the following keywords (in order of checking):
+     *
+     * - **--version** Prints the version information and throws 
```

Comment:
![ugly_font](https://user-images.githubusercontent.com/7955878/28411119-201cd804-6d40-11e7-88a8-207eab6d8164.png)

...rather no :no_good_man: 
(there's hardly a difference to the normal font)